### PR TITLE
support SubjectAlternativeNames

### PR DESCRIPTION
**Pull request by sawanoboly**
Hi, thank you for mail. As I reported, it is still a work in progress.

Done:
- create new attribute `alt_names`
- include alt_names into CSR.

TODO:
- verify all domains by HTTP01.

----

F.Y.I. Token is different all for each domain. We have to do verify before send CSR to the boulder.

For example, there are domains such as `example.com` as common_name, `www.example.com` and `mail.example.com` as alternative names.
We have to create a challenge to all of them, and we must prove the reachability.

### CODE REVIEW

**Comment by sawanoboly** [comment]
@thoutenbos 

I've implemented. Please check it.

File: libraries/acme.rb [hunk 1]
```diff
@@ -64,14 +64,31 @@ def acme_validate(authz)
...
+    break unless validate.verify_status == 'pending'
+    times += 1
+    sleep 10
+  end
+
+  validate
+end
+
+def acme_cert(cn, key, alt_names = [])
+  csr = OpenSSL::X509::Request.new
```
Comment (thoutenbos) [inline libraries/acme.rb]: I think this line might be obsolete now?

File: providers/certificate.rb [hunk 2]
```diff
@@ -52,89 +52,92 @@
...
+      authz = acme_authz domain
 
-    when 'pending'
-      case new_resource.method
-      when 'http'
-        tokenpath = "#{new_resource.wwwroot}/#{authz.http01.filename}"
+      case authz.status
+      when 'valid'
+        authz
+        next
```
Comment (thoutenbos) [inline providers/certificate.rb]: Why the `next` here? It should be ready to request the cert now

**Comment by thoutenbos** [comment]
Very nice work!

File: libraries/acme.rb (hunk 1 above)
Comment (sawanoboly) [inline libraries/acme.rb]: Ah, yes. I/ve forgotten remove. I'll update.

File: providers/certificate.rb (hunk 2 above)
Comment (sawanoboly) [inline providers/certificate.rb]: Lines which after `next` will be removed. I've forgotten remove too.
Then, we have to check that all domains are validated when create certificate with SANs to make sure.

Comment (sawanoboly) [inline providers/certificate.rb]: I noticed that in case `when 'valid'` doesn't returns validation. I'll update, please wait.

**Comment by sawanoboly** [comment]
I've cleaned up.

**Comment by thoutenbos** [comment]
Great!

**Pull request merged**